File: algorithms/1_easy/C/01941_check_if_all_characters_have_equal_number_of_occurrences/solution.cpp
class Solution {
public:
    static bool areOccurrencesEqual(const string& s) {
        if (s.empty()) { return true; }
        size_t freqs[256] = { 0 };

        for (const unsigned char ch : s) {
            freqs[ch] += 1;
        }

        const size_t freq_first = freqs[static_cast<unsigned char>(s[0])];
        for (size_t i = 0; i < 256; i += 1) {
            if (freqs[i] != 0 && freqs[i] != freq_first) {
                return false;
            }
        }

        return true;
    }
};

File: algorithms/1_easy/C/01941_check_if_all_characters_have_equal_number_of_occurrences/solution.rs
impl Solution {
    pub fn are_occurrences_equal(s: String) -> bool {
        let mut freqs = [0_usize; 256];

        for &b in s.as_bytes() {
            freqs[b as usize] += 1;
        }

        if let Some(&val) = freqs.iter().filter(|&&x| x != 0).next() {
            for f in freqs.into_iter().filter(|&x| x != 0) {
                if f != val {
                    return false;
                }
            }
        }

        true
    }
}
